projects/project-2/assignment/Assginment 2.md

```diff
@@ -0,0 +1,241 @@
+<header>
+  <h1>Project 2: The SPARQL Library of Common Core Ontologies</h1>
+</header>
+
+1. Detecting Cycles in the Hierarchy: This query can help identify problematic cyclic subclass relationships in an ontology, which often signal modeling errors or unintended structures. 
+(level 6 - 3 pts)
+
+<pre>
+  <code>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+SELECT ?classA ?classB
+WHERE {
+  ?classA rdfs:subClassOf+ ?classB .
+  ?classB rdfs:subClassOf+ ?classA .
+  FILTER(?classA != ?classB)  # Exclude trivial cycles where classA and classB are the same
+}
+</code>
+</pre>
+
+
+2. Detecting missing labels: This version splits the `rdfs:label` and `rdfs:comment` checks into two separate sections, each with its own `FILTER` and `BIND` statements,
+ making it clear whether a term is missing a label or a comment. (level 4 - 10 pts)
+
+<pre>
+  <code>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+
+SELECT ?term ?missingAnnotation
+WHERE {
+    {
+        # Find all terms missing rdfs:label
+        {
+            ?term a rdfs:Class .
+        }
+        UNION
+        {
+            ?term a rdf:Property .
+        }
+        UNION
+        {
+            ?term a owl:NamedIndividual .
+        }
+        
+        FILTER NOT EXISTS { ?term rdfs:label ?label . }
+        BIND("Missing rdfs:label" AS ?missingAnnotation)
+    }
+    UNION
+    {
+        # Find all terms missing rdfs:comment
+        {
+            ?term a rdfs:Class .
+        }
+        UNION
+        {
+            ?term a rdf:Property .
+        }
+        UNION
+        {
+            ?term a owl:NamedIndividual .
+        }
+
+        FILTER NOT EXISTS { ?term rdfs:comment ?comment . }
+        BIND("Missing rdfs:comment" AS ?missingAnnotation)
+    }
+}
+</code>
+</pre>
+
```

Comment:
I would check the query again bc I´m getting an error when I execute it on Protégé

Reply:
Oh okay! Thanks for checking!